scripts/python/checks/validate_retry_usage.py

```diff
@@ -0,0 +1,109 @@
+import ast
+import sys
+
+
+class RetryValidator(ast.NodeVisitor):
+    def __init__(self, filename):
+        self.errors = []
+        self.filename = filename
+        self.processed_lines = set()  # Track processed lines
+
+    def visit_Call(self, node):
+        """Visit function calls and check for retry usage."""
+        if isinstance(node.func, ast.Name) and node.func.id == "retry":
+            self._validate_retry_call(node)
+        self.generic_visit(node)
+
+    def visit_FunctionDef(self, node):
+        """Visit function definitions and check for retry decorator."""
+        for decorator in node.decorator_list:
+            if (
+                isinstance(decorator, ast.Call)
+                and isinstance(decorator.func, ast.Name)
+                and decorator.func.id == "retry"
+            ):
+                self._validate_retry_call(decorator)
+        self.generic_visit(node)
+
+    def _validate_retry_call(self, node):
+        """Validate the arguments of a retry call."""
+        if node.lineno in self.processed_lines:
+            return  # Skip already processed lines
+        self.processed_lines.add(node.lineno)
+        kwargs = {arg.arg: arg.value for arg in node.keywords}
+
+        # Retrieve and evaluate values
+        tries = self._eval_value(kwargs.get("tries"), default=4)
+        delay = self._eval_value(kwargs.get("delay"), default=3)
+        backoff = self._eval_value(kwargs.get("backoff"), default=2)
+        max_delay = self._eval_value(kwargs.get("max_delay"), default=600)
+        max_timeout = self._eval_value(kwargs.get("max_timeout"), default=14400)
+
+        # Calculate total retry time
+        total_time = self._calculate_total_time(tries, delay, backoff, max_delay)
+
+        # Validation rules
+        if tries is not None and tries <= 0:
+            self.errors.append((node.lineno, "Invalid 'tries': must be > 0."))
+        if delay is not None and delay <= 0:
+            self.errors.append((node.lineno, "Invalid 'delay': must be > 0."))
+        if backoff is not None and backoff < 1:
+            self.errors.append((node.lineno, "Invalid 'backoff': must be >= 1."))
+        if max_delay is not None and max_delay <= 0:
+            self.errors.append((node.lineno, "Invalid 'max_delay': must be > 0."))
+        if max_timeout is not None and max_timeout <= 0:
+            self.errors.append((node.lineno, "Invalid 'max_timeout': must be > 0."))
+        if total_time > max_timeout:
+            self.errors.append(
+                (
+                    node.lineno,
+                    f"Total retry time ({total_time}s or {total_time // 3600}h {total_time % 3600 // 60}m) "
+                    f"exceeds max_timeout ({max_timeout}s or {max_timeout // 3600}h {max_timeout % 3600 // 60}m).",
+                )
+            )
+
+    def _calculate_total_time(self, tries, delay, backoff, max_delay):
```

Comment:
I assume it won't be straight forward to have a single implementation for this and the one in [ocs_ci/utility/retry.py](https://github.com/red-hat-storage/ocs-ci/pull/11224/files#diff-8853bdf565bbbd80748a64aeff9cc34510945363a6c88e4cbc89b81bd1ae6b8f), right?

Reply:
This is simple script located in this path and completely independent on the framework. If it will require imports from framework it will require to install whole virtual env of ocs-ci which is not worth for this purpose and I would rather keep it separate here.